### PR TITLE
Auth controller

### PR DESCRIPTION
**Pull request by jenow**
## What does this PR do ?

Add auth controller with unit tests and documentation.

### How should this be manually tested?

```sh
./gradlew test
```

### Other changes

- Added test snippet documentation
- Reformat code with new prettier rules

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,6 @@ bin
 bin
 .classpath
 .project
-.settings
```
Comment (alexandrebouthinon) [inline .gitignore]: Why did you remove that folder from the gitignore?

File: .ci/doc/templates/catch.tpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import io.kuzzle.sdk.Kuzzle;
```
Comment (alexandrebouthinon) [inline .ci/doc/templates/catch.tpl.java]: I didn't get the purpose of this snippet template

File: doc/3/controllers/auth/create-my-credentials/index.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
+code: true
+type: page
+title: CreateMyCredentials
+description: Create the current user's credentials for the specified strategy.
+---
+
+# CreateMyCredentials
+
+Create the current user's credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/create-my-credentials/index.md]:
```suggestion
Creates the current user's credentials for the specified strategy.
```

File: doc/3/controllers/auth/create-my-credentials/snippets/create-my-credentials.test.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+name: Auth#CreateMyCredentials
+description: Create the current user's credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/create-my-credentials/snippets/create-my-credentials.test.yml]:
```suggestion
description: Creates the current user's credentials for the specified strategy.
```

File: doc/3/controllers/auth/credentials-exist/index.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
+code: true
+type: page
+title: CredentialsExist
+description: Check that the current user has credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/credentials-exist/index.md]:
```suggestion
description: Checks that the current user has credentials for the specified strategy.
```

File: doc/3/controllers/auth/credentials-exist/snippets/credentials-exist.test.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name: Auth#CredentialsExistAsync
+description: Check that the current user has credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/credentials-exist/snippets/credentials-exist.test.yml]:
```suggestion
description: Checks that the current user has credentials for the specified strategy.
```

File: doc/3/controllers/auth/update-my-credentials/index.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+code: true
+type: page
+title: UpdateMyCredentials
+description: Update the current user's credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/update-my-credentials/index.md]:
```suggestion
description: Updates the current user's credentials for the specified strategy.
```

File: doc/3/controllers/auth/update-my-credentials/index.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+# UpdateMyCredentials
+
+Update the current user's credentials for the specified strategy. The credentials to send will depend on the authentication plugin and the authentication strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/update-my-credentials/index.md]:
```suggestion
Updates the current user's credentials for the specified strategy. The credentials to send will depend on the authentication plugin and the authentication strategy.
```

File: doc/3/controllers/auth/validate-my-credentials/index.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+---
+code: true
+type: page
+title: ValidateMyCredentials
+description: Validate the current user's credentials for the specified strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/validate-my-credentials/index.md]:
```suggestion
description: Validates the current user's credentials for the specified strategy.
```

File: doc/3/controllers/auth/validate-my-credentials/index.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+# ValidateMyCredentials
+
+Validate the current user's credentials for the specified strategy. The `result` field is `true` if the provided credentials are valid; otherwise an error is triggered. This route does not actually create or modify the user credentials. The credentials to send will depend on the authentication plugin and authentication strategy.
```
Comment (alexandrebouthinon) [inline doc/3/controllers/auth/validate-my-credentials/index.md]:
```suggestion
Validates the current user's credentials for the specified strategy. The `result` field is `true` if the provided credentials are valid; otherwise an error is triggered. This route does not actually create or modify the user credentials. The credentials to send will depend on the authentication plugin and authentication strategy.
```

File: doc/3/core-classes/kuzzle/disconnect/index.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+---
+code: true
+type: page
+title: Disconnect
+description: Connects the SDK to Kuzzle
```
Comment (alexandrebouthinon) [inline doc/3/core-classes/kuzzle/disconnect/index.md]:
```suggestion
description: Disconnects the SDK from Kuzzle
```

File: doc/3/core-classes/kuzzle/disconnect/index.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+# Disconnect
+
+Closes the current connection to Kuzzle.
+
+Disconnect the SDK from the Kuzzle server using the subsequent protocol `Disconnect` method.
```
Comment (alexandrebouthinon) [inline doc/3/core-classes/kuzzle/disconnect/index.md]:
```suggestion
Disconnects the SDK from the Kuzzle server using the subsequent protocol `Disconnect` method. 
```

File: .ci/doc/templates/catch.tpl.java (hunk 2 above)
Comment (jenow) [reply .ci/doc/templates/catch.tpl.java]: Used by the `createMyCredentials` snippet, since there is only one authentication plugin in the Kuzzle used to test, there is only one strategy so this method is supposed to throw a io.kuzzle.sdk.Exceptions.ApiErrorException

File: .gitignore (hunk 1 above)
Comment (jenow) [reply .gitignore]: Because I've set tons of prettier rules inside so everyone uses the same.

File: src/test/java/io/kuzzle/test/CoreClasses/TaskTest/TaskTests.java [hunk 13]
```diff
@@ -13,7 +13,7 @@
 import static org.mockito.Mockito.*;
 
 public class TaskTests {
-
+/*
```
Comment (scottinet) [inline src/test/java/io/kuzzle/test/CoreClasses/TaskTest/TaskTests.java]: remove commented code

File: doc/3/core-classes/kuzzle/connect/index.md [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+## Arguments
+
+```java
+public void connect() throws Exception;
+```
+
+## Throws
+
+Throws an `Exception`.
```
[the remark this answers is not in the document]
Comment (jenow) [reply doc/3/core-classes/kuzzle/connect/index.md]: This is because it's an interface so it depends how you implement connect (with which library etc) and it could throw any exception, even custom exceptions which are part of a library

File: doc/3/controllers/auth/update-my-credentials/index.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+| Argument      | Type               | Description                          |
+|---------------|--------------------|--------------------------------------|
+| `strategy`    | <pre>String</pre>  | Strategy to use                      |
+| `credentials` | <pre>ConcurrentHashMap<String, Object></pre> | JObject representing the credentials |
+
+## Throws
+
+A `NotConnectedException`, `InternalException` if there is an error.
```
[the remark this answers is not in the document]
Comment (jenow) [reply doc/3/controllers/auth/update-my-credentials/index.md]: I agree with you, it's not really useful
